test/integration/vpc_e2e_test.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2022 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"path"
+	"reflect"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"antrea.io/nephe/apis/crd/v1alpha1"
+	cloudUtils "antrea.io/nephe/pkg/cloud-provider/utils"
+	"antrea.io/nephe/test/utils"
+)
+
+var _ = Describe(fmt.Sprintf("%s,%s: VPC Inventory", focusAws, focusAzure), func() {
+	var (
+		namespace     *v1.Namespace
+		vmIDs         []string
+		vpcID         string
+		vpcName       string
+		cloudProvider string
+		kind          string
+		vpcObjectName string
+	)
+
+	createNS := func() {
+		logf.Log.Info("Create", "namespace", namespace.Name)
+		err := k8sClient.Create(context.TODO(), namespace)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	deleteNS := func() {
+		logf.Log.Info("Delete", "namespace", namespace.Name)
+		err := k8sClient.Delete(context.TODO(), namespace)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	verifyVpcInventory := func(vpcObjectName string, vpcID string) error {
+		err := wait.Poll(time.Second*5, time.Second*30, func() (bool, error) {
+			// Find if vpc with expected Id is present in the vpc inventory.
+			cmd := fmt.Sprintf("get vpc %v -n %v -o json -o=jsonpath={.spec.Id}", vpcObjectName, namespace.Name)
+			out, err := kubeCtl.Cmd(cmd)
+			logf.Log.V(1).Info("Got VPC", "VpcID", out)
+			if err != nil {
+				logf.Log.V(1).Info("Failed to get VPC", "VPC object name", vpcObjectName, "err", err)
+				return false, nil
+			}
+			Expect(out).To(Equal(vpcID))
+			return true, nil
+		})
+		return err
+	}
+
+	verifyNoVpcInventory := func() error {
```

Comment:
verifyNoVPC and VerifyVPc share most of the code. Consider making them 1 function and use if else